sshfs/spec.py

```diff
@@ -54,6 +56,19 @@ def __init__(
 
         super().__init__(self, **kwargs)
 
+        _sftp_client_args = {
+            k: kwargs.pop(k)
+            for k in kwargs.copy().keys()
+            if k
+            in {
+                "env",
+                "send_env",
+                "path_encoding",
+                "path_errors",
+                "sftp_version",
+            }
+        }
```

Comment:
Seems a bit strange to be doing this twice (also this will silently ignore any unrecognised options). How about just creating an explicit `sftp_client_kwargs` parameter and then just pass it down like that as a whole?

Reply:
I wanted to keep the "any kwargs will be passed to underlying library" behaviour but it is a bit awkward. I guess I could add explicit parameters for sftp_client. Should I also make explicit parameter for asyncssh.connect params?

Reply:
From a quick look doesn't seem like you really need to touch `asyncssh.connect` params, so I would go from the need and start with `sftp_client_kwargs`

Reply:
Removed that part in favour of sftp_client_kwargs parameter